src/omotes_sdk/workflow_type.py

```diff
@@ -1,5 +1,116 @@
+import json
 from dataclasses import dataclass, field
-from typing import List, Optional, Dict
+from datetime import datetime
+from typing import List, Optional, Dict, Union
+from typing_extensions import Self
+
+from omotes_sdk_protocol.workflow_pb2 import (
+    AvailableWorkflows,
+    Workflow,
+    WorkflowParameter as WorkflowParameterPb,
+    StringParameter as StringParameterPb,
+    StringEnum as StringEnumPb,
+    BooleanParameter as BooleanParameterPb,
+    IntegerParameter as IntegerParameterPb,
+    FloatParameter as FloatParameterPb,
+    DateTimeParameter as DateTimeParameterPb,
+)
+
+
+@dataclass(eq=True, frozen=True)
+class WorkflowParameter:
+    """Define a workflow parameter this SDK supports."""
+
+    key_name: str = field(hash=True, compare=True)
+    """Key name for the parameter."""
+    title: str | None = field(default=None, hash=True, compare=True)
+    """Optionally override the 'snake_case to text' 'key_name' (displayed above the input field)."""
+    description: str | None = field(default=None, hash=True, compare=True)
+    """Optional description (displayed below the input field)."""
+
+    def __post_init__(self) -> None:
+        """Check parameter format."""
+        for name, field_type in self.__annotations__.items():
+            # Do not check dataclasses (like KeyDisplayPair)
+            # TODO better way of checking that 'field_type' is dataclass?
```

Comment:
The check at runtime is needed due to the configuration from json file.  
I looked into pydantic but then we would need to adapt several things, for instance `__eq__` and `__hash__`. Overwriting `__set__` would also add much more code. Do you think it is worth it?

Reply:
I am okay leaving it as is, except for removing the TODO as this would then be the accepted way of handling it. However, if you feel we need to improve it in some direction, I do think any of the options above would work. Specifically `__eq__` and `__hash__` are already generated by the dataclasses.

Reply:
If we would use pydantic, which is the approach I would prefer, we don't have the option to leave parameters out of hash and equality like so: `default: str | None = field(default=None, hash=False, compare=False)`. So then it would be necessary to overwrite  `__eq__` and `__hash__`.  
I will remove the TODO and leave it as is.